modules/pipeline/pipengine/actionexecutor/manager.go

```diff
@@ -73,11 +95,54 @@ func (m *Manager) Get(name types.Name) (types.ActionExecutor, error) {
 	if len(name) == 0 {
 		return nil, errors.Errorf("executor name is empty")
 	}
+	m.RLock()
 	e, ok := m.executors[name]
+	m.RUnlock()
+	if ok {
+		return e, nil
+	}
+	m.RLock()
+	kind, ok := m.kinds[name]
+	m.RUnlock()
+	if !ok {
+		// if the executor is not found, it may be a k8s type executor
+		// so try to get kind from name and store in the kinds cache
+		var found bool
+		kind, found = m.GetK8sKindByExecutorName(name)
```

Comment:
check k8s kind here is also bad.

Reply:
adjusted, use `GetKindByExecutorName` get all kind
![image](https://user-images.githubusercontent.com/30427474/158321414-977836c3-9dc0-4d43-b088-870b84692247.png)